packages/solarwinds-apm/src/index.es.ts

```diff
@@ -14,18 +14,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-module.exports.callsite = function callsite() {
-  const prepareStackTrace = Error.prepareStackTrace
-  try {
-    const callsites = []
-    Error.prepareStackTrace = (_err, cs) => {
-      callsites.push(...cs)
-    }
-    void new Error().stack
-
-    const current = callsites[0]
-    return callsites.find((cs) => cs.getFileName() !== current.getFileName())
-  } finally {
-    Error.prepareStackTrace = prepareStackTrace
-  }
+import module from "node:module"
+import process from "node:process"
+
+import semver from "semver"
+
+import { init } from "./init.js"
+import { setter } from "./symbols.js"
+
+try {
+  await init()
+} catch (err) {
+  console.warn(err)
 }
+
+const setRegister = setter("register")
+if (setRegister && semver.gte(process.versions.node, "20.8.0")) {
+  setRegister()
+  module.register("./hooks.es.js", import.meta.url)
+}
```

Comment:
The `--loader` flag is deprecated in Node >=20.8 in favour of this new function.